utils.go

```diff
@@ -19,12 +19,14 @@ func ConfigFromConnString(in string) (*Config, error) {
 			return nil, fmt.Errorf("invalid connection string: %s (%s)", in, err.Error())
 		}
 		fields := strings.Split(strings.TrimPrefix(u.Path, "/"), "/")
-		if len(fields) != 2 {
+		if len(fields) == 2 {
+			cfg.DatasetID = fields[1]
+		}
+		if len(fields) > 2 {
 			return nil, fmt.Errorf("invalid connection string: %s", in)
 		}
```

Comment:
What if the connection string is of length less than 2?

Reply:
less than 2 and not nil will be 1, and it'll get assigned to fields[0] variable